File: src/easy/RemoveElement.go
package main

func removeElement(nums []int, val int) int {
	index, replaceIndex := 0, len(nums)-1
	for index <= replaceIndex {
		if nums[index] == val {
			nums[index] = nums[replaceIndex]
			replaceIndex--
		} else {
			index++
		}
	}
	return index
}

File: src/easy/RemoveElement.java
public class RemoveElement {
    public int removeElement(int[] nums, int val) {
        int tail = nums.length - 1;
        int current = 0;
        while (current <= tail) {
            if (nums[current] != val) {
                current++;
                continue;
            }

            int temp = nums[current];
            nums[current] = nums[tail];
            nums[tail] = temp;
            tail--;
        }
        return tail + 1;
    }
}
